src/runtime/GlobalObjectBuiltinRegExp.cpp

```diff
@@ -641,36 +647,79 @@ static Value builtinRegExpUnicodeGetter(ExecutionState& state, Value thisValue,
     return builtinRegExpOptionGetterHelper(state, thisValue, RegExpObject::Option::Unicode);
 }
 
+static Value getLegacyRegExpStaticProperty(ExecutionState& state, Value thisValue, Value stringVal)
+{
+    if (!thisValue.isPointerValue() || thisValue.asPointerValue() != state.context()->globalObject()->regexp()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "thisValue is not RegExp");
+    }
+    if (stringVal.isEmpty()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "Value is empty");
+    }
+    return stringVal;
+}
+
+static Value setLegacyRegExpStaticProperty(ExecutionState& state, Value thisValue, Value stringVal, StringView* strView)
+{
+    if (!thisValue.isPointerValue() || thisValue.asPointerValue() != state.context()->globalObject()->regexp()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "thisValue is not RegExp");
+    }
+    strView = new StringView(stringVal.toString(state));
+    return Value();
+}
+
 // For non-standard, read-only properties of RegExp
 static Value builtinRegExpInputGetter(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
 {
-    return state.resolveCallee()->codeBlock()->context()->regexpStatus().input;
+    return getLegacyRegExpStaticProperty(state, thisValue, state.resolveCallee()->codeBlock()->context()->regexpStatus().input);
 }
 
 static Value builtinRegExpInputSetter(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
 {
+    if (!thisValue.isPointerValue() || thisValue.asPointerValue() != state.context()->globalObject()->regexp()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, "thisValue is not RegExp");
+    }
     state.resolveCallee()->codeBlock()->context()->regexpStatus().input = argv[0].toString(state);
     return Value();
 }
 
 static Value builtinRegExpLastMatchGetter(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
 {
-    return Value(new StringView(state.resolveCallee()->codeBlock()->context()->regexpStatus().lastMatch));
+    return getLegacyRegExpStaticProperty(state, thisValue, Value(new StringView(state.resolveCallee()->codeBlock()->context()->regexpStatus().lastMatch)));
+}
+
+static Value builtinRegExpLastMatchSetter(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
```

Comment:
According to the standard, `RegExp.lastMatch` does not have setter (getter only).
Remove this setter function and other setter functions too.